src/types/UserChangemakerRole.ts

```diff
@@ -0,0 +1,39 @@
+import { ajv } from '../ajv';
+import { AccessType } from './AccessType';
+import type { JSONSchemaType } from 'ajv';
+import type { Writable } from './Writable';
+import type { KeycloakUserId } from './KeycloakUserId';
+
+interface UserChangemakerRole {
+	readonly userKeycloakUserId: KeycloakUserId;
+	readonly accessType: AccessType;
+	readonly changemakerId: number;
```

Comment:
Nice. I take this order to be the subject-permission-object scheme I was looking for. I don't think of this as a role but I suppose it could be thought of as a role. Usually I think of role as the hat one wears in an organization, such as "I do software engineering: I have a role of software engineer" and "I also administer some systems: I also have a role of system administrator." In other words, roles are associated with people, not with the objects in the access control. After the roles are established there would be a separate modeling of  permissions on objects granted to certain roles.

Reply:
Very true -- I think the word `role` made sense originally when we were thinking in terms of `administrator, editor, viewer` but when we switched to actions (aka permissions) the word should have changed.  I agree with your naming suggestion in an earlier comment.